### DISCUSSION TITLE
Provide training data and output folder for rendering images as cmd line argument

### DISCUSSION THREAD

**Discussion by MrNeRF** [JamesPerlman/TurboNeRF#6] [Ideas]
Hi JPerlDev, 
are you interested to implement the input data source and the rendering output destination as cmd line arguments?
Right now I am experimenting with your implementation and I on every commit I have to update the source code with my folder locations. Any interest in this?

Best
Janusch

**Comment by JamesPerlman** [comment]
Hey Janusch!  This is a great idea.  What do you think about using the python.api instead of main.cu?  I'd assume it's easier to create and maintain a command-line tool in python than it will be in CUDA.  Check out [examples/train.py](https://github.com/JamesPerlman/NeRFRenderCore/blob/main/examples/train.py) and lmk what you think.

**Reply by MrNeRF** [reply]
Your project is very interesting and your progress is fantastic. Right know I am more interested to learn CUDA as deep as I can (I mean the rather advanced optimization tricks). Since your project is very interesting, my hope is that I will be able to contribute with some optimizations in near future in the CUDA part. For now, the python interface is not of that much interest to me.

The paths, where the training data is located are hard-coded and reflect your system layout. This holds also true for the rendering output path. I could maintain a branch with my specific settings, but then I have to always merge the master into my branch and resolve the conflicts. I would suggest a solution, that defaults to your settings but that also allows to specify custom paths. Let me suggest an implementation and the you can still decided if you like it. Its not that much work.

**Reply by JamesPerlman** [reply]
Ok!  I don't have experience with writing a CUDA or C++ command line tool - if you come up with a good mechanism then I'll absolutely approve it.  Thanks!

**Reply by JamesPerlman** [reply]
Also I just want to say I really appreciate your interest in this project and I absolutely love the idea of hyper-optimizing every last detail using some advanced techniques.  Cheers 😄

**Reply by MrNeRF** [reply]
Nice, thx.  Although I code professionally in C++ I don't have any CUDA experience especially not in such depth. This is supposed to be my pet learning endeavor after work. Unfortunately, I can't invest every evening so I won't promise magical things will happen. But lets see... At least I can help that it compiles on Linux on Ubuntu on a regular basis.
